lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -457,6 +498,137 @@ def _check_job_arguments(self, jobs, has_parent_job=False):
             # although most likely jobs aren't operating on the same object
             self._check_ws_objects(source_objects=job.get(_SOURCE_WS_OBJECTS))
 
+    def retry_multiple(
+        self, job_ids, as_admin=False
+    ) -> List[Dict[str, Union[str, Any]]]:
+        """
+        #TODO Add new job requirements/cgroups as an optional param
+        :param job_ids: The list of jobs to retry
+        :param as_admin: Run with admin permission
+        :return: The child job ids that have been retried or errors
+        """
+        retried_jobs = []
+        for job_id in job_ids:
+            try:
+                retried_jobs.append(self.retry(job_id, as_admin=as_admin))
+            except Exception as e:
+                retried_jobs.append({"job_id": job_id, "error": f"{e}"})
+        return retried_jobs
+
+    @staticmethod
+    def _retryable(status: str):
+        return status in [Status.terminated.value, Status.error.value]
+
+    def retry(self, job_id: str, as_admin=False) -> Dict[str, Optional[str]]:
+        """
+        #TODO Add new job requirements/cgroups as an optional param
+        :param job_id: The main job to retry
+        :param as_admin: Run with admin permission
+        :return: The child job id that has been retried
+        """
+        job = self.sdkmr.get_job_with_permission(
+            job_id, JobPermissions.WRITE, as_admin=as_admin
+        )  # type: Job
+
+        # TODO: Fix this when updating the run_job_batch endpoint
+        # if job.batch_job:
+        #     raise CannotRetryJob("Cannot retry parent container jobs")
```

Comment:
I don't understand why this check needs to be commented out

Reply:
Because all jobs launched by the batch uploader are batch_job=True

Reply:
? The child jobs aren't marked as `batch_job=True` surely? I don't see that happening in the code - only the parent job is marked.

Reply:
Ah I was thinking that we would use this field to differentiate between batch jobs and kbparallels jobs, but as it stands, I guess that isn't happening, so I will put it back in

Reply:
:+1: